app/src/main/java/com/incwelltechnology/lms/ui/auth/LoginActivity.kt

```diff
@@ -0,0 +1,41 @@
+package com.incwelltechnology.lms.ui.auth
+
+import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.incwelltechnology.lms.R
+import com.incwelltechnology.lms.databinding.ActivityLoginBinding
+import com.incwelltechnology.lms.util.toast
+
+class LoginActivity : AppCompatActivity(), AuthListener {
+    val TAG = LoginActivity::class.java.simpleName
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val binding: ActivityLoginBinding = DataBindingUtil.setContentView(this, R.layout.activity_login)
+        val viewModel = ViewModelProviders.of(this).get(AuthViewModel::class.java)
+        binding.viewmodel = viewModel
+        viewModel.authListener = this
+    }
+
+    override fun onStarted() {
+        //progress_bar.show()
+    }
+
+    override fun onSuccess(loginResponse: LiveData<String>) {
+        loginResponse.observe(this, Observer {
```

Comment:
@iamsannyrai you are observing the onSuccess only which only observes once you called the success.

Reply:
yes because for other function we are providing normal message(string type)